fog/ledger/enclave/impl/src/oblivious_utils.rs

```diff
@@ -0,0 +1,276 @@
+        assert_eq!(results, shard_results);
+    }
+
+    #[test]
+    fn duplicate_shard_results_returns_one_result() {
+        let client_queries = vec![KeyImageQuery {
+            key_image: 1.into(),
+            start_block: 0,
+        }];
+        let key_image_result = KeyImageResult {
+            key_image: 1.into(),
+            spent_at: 2,
+            timestamp: 3,
+            timestamp_result_code: TimestampResultCode::WatcherBehind as u32,
+            key_image_result_code: KeyImageResultCode::Spent as u32,
+        };
+        let shard_results = vec![key_image_result.clone(), key_image_result.clone()];
+        let mut results = collate_shard_key_image_search_results(client_queries, &shard_results);
+        results.sort_by_key(|r| r.key_image);
+        assert_eq!(results, vec![key_image_result]);
+    }
+
+    #[test]
+    fn none_available() {
+        let client_queries = vec![KeyImageQuery {
+            key_image: 1.into(),
+            start_block: 0,
+        }];
+
+        let shard_results = vec![];
+
+        let result = collate_shard_key_image_search_results(client_queries, &shard_results);
+
+        assert_eq!(result, vec![default_client_key_image(1.into())]);
+    }
+
+    #[test]
+    fn error_result() {
+        let client_queries = vec![KeyImageQuery {
+            key_image: 1.into(),
+            start_block: 0,
+        }];
+
+        let key_image_result = KeyImageResult {
+            key_image: 1.into(),
+            spent_at: 1,
+            timestamp: 123,
+            timestamp_result_code: TimestampResultCode::TimestampFound as u32,
+            key_image_result_code: KeyImageResultCode::KeyImageError as u32,
+        };
+        let shard_results = vec![key_image_result.clone()];
+
+        let results = collate_shard_key_image_search_results(client_queries, &shard_results);
+
+        assert_eq!(results, vec![key_image_result]);
+    }
+
+    #[test]
+    fn partial_responses() {
+        let client_queries = vec![
+            KeyImageQuery {
+                key_image: 1.into(),
+                start_block: 0,
+            },
+            KeyImageQuery {
+                key_image: 2.into(),
+                start_block: 0,
+            },
+        ];
+
+        let key_image_result = KeyImageResult {
+            key_image: 1.into(),
+            spent_at: 1,
+            timestamp: 123,
+            timestamp_result_code: TimestampResultCode::TimestampFound as u32,
+            key_image_result_code: KeyImageResultCode::Spent as u32,
+        };
+        let shard_results = vec![key_image_result];
+
+        let results = collate_shard_key_image_search_results(client_queries, &shard_results);
+
+        assert_eq!(
+            results[0].key_image_result_code,
+            KeyImageResultCode::Spent as u32
+        );
+        assert_eq!(
+            results[1].key_image_result_code,
+            DEFAULT_KEY_IMAGE_SEARCH_RESULT_CODE as u32
+        );
```

Comment:
:wrench: Consider the following to more explicitly ensure only 2 entries in the results and that all the fields were populated.  For the all the fields correctly, I don't think it's necessary since other tests more or less checked that, but it keeps this test consistent with the others.
```suggestion
        let shard_results = vec![key_image_result.clone()];

        let mut results = collate_shard_key_image_search_results(client_queries, &shard_results);
        results.sort_by_key(|r| r.key_image);
        assert_eq!(
            results,
            vec![key_image_result, default_client_key_image(2.into())]
        );
```